src/Arcus.Observability.Telemetry.AspNetCore/Extensions/ILoggerHttpDependencyExtensions.cs

```diff
@@ -41,8 +41,8 @@ public static class ILoggerHttpDependencyExtensions
             Guard.NotNull(logger, nameof(logger), "Requires a logger instance to track telemetry");
             Guard.NotNull(request, nameof(request), "Requires a HTTP request message to track a HTTP dependency");
             Guard.NotNull(measurement, nameof(measurement), "Requires a dependency measurement instance to track the latency of the HTTP communication when tracking a HTTP dependency");
-            Guard.For(() => !Enum.IsDefined(typeof(HttpStatusCode), statusCode),
-                new ArgumentException("Requires a response HTTP status code that's within the bound of the enumeration to track a HTTP dependency"));
+            Guard.NotLessThan((int)statusCode, 100, nameof(statusCode), "Requires a valid HTTP response status code that's within the range of 100 to 599, inclusive");
```

Comment:
Thank you for your contribution!

I was thinking that we could be facing an InvalidCastException here, but it seems like that is not the case.  
(HttpStatusCode)599 for instance just works.  (This is also something that is covered by the tests).

Although, I wonder if we should have an overload that accepts an `int statusCode` to make this more explicit.  To be honest, it would not come up in my mind to just cast an arbitrary integer to a HttpStatusCode enum value, as I would have expected that this would lead to an invalidcast exception.

Reply:
Yes, I agree here. We have also `LogRequest` overloads that take in both `HttpStatusCode` and `int`. So it would be good to have this here, too.